pallets/ocex/src/lib.rs

```diff
@@ -391,6 +393,55 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Updates the trading pair config
+		#[pallet::weight(100000)]
+		pub fn update_trading_pair(
+			origin: OriginFor<T>,
+			base: AssetId,
+			quote: AssetId,
+			min_order_price: BalanceOf<T>,
+			max_order_price: BalanceOf<T>,
+			min_order_qty: BalanceOf<T>,
+			max_order_qty: BalanceOf<T>,
+			price_tick_size: BalanceOf<T>,
+			qty_step_size: BalanceOf<T>,
+		) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+			ensure!(base != quote, Error::<T>::BothAssetsCannotBeSame);
+			ensure!(
+				<TradingPairs<T>>::contains_key(base, quote),
+				Error::<T>::TradingPairNotRegistered
+			);
+			let trading_pair_info = TradingPairConfig {
+				base_asset: base,
+				quote_asset: quote,
+				min_price: Decimal::from(min_order_price.saturated_into::<u128>())
```

Comment:
This will overflow.

Reply:
Now ensuring that these params do not exceed `10^27`

Reply:
You also need to check for these if values are not zero

Reply:
Updated ✅